File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/rest/RestExecutorFactory.java
package co.uk.swarmbit.rest;

public class RestExecutorFactory {

    public static RestExecutor createRestExecutor(RestMethod method) {
        switch (method) {
            case POST:
                return new PostExecutor();
            case DELETE:
                return new DeleteExecutor();
            default:
                return new GetExecutor();
        }
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/common/rest/executor/RestExecutorFactory.kt
package com.swarmbit.docker.api.common.rest.executor

import com.swarmbit.docker.api.common.rest.model.Method
import com.swarmbit.docker.api.common.rest.model.Method.DELETE
import com.swarmbit.docker.api.common.rest.model.Method.POST

object RestExecutorFactory {
    fun createRestExecutor(method: Method): RestExecutor {
        return when (method) {
            POST -> PostExecutor()
            DELETE -> DeleteExecutor()
            else -> GetExecutor()
        }
    }
}